.github/workflows/ci.yml

```diff
@@ -0,0 +1,70 @@
+name: ci-pr
+
+on:
+  pull_request:
+  workflow_dispatch:
+  push:
+    branches: [main]
+
+jobs:
+  betree-build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        profile: minimal
+        override: true
+    - name: Dependency Cache
+      uses: Swatinem/rust-cache@v1
```

Comment:
Can we easily replace this (and `actions-rs/toolchain` above) with "official" actions (such as https://github.com/actions/cache) or is that too much work?

Reply:
It's fine to exchange the `rust-cache` with the official cache action. I'll have a look if the pre- and post-treatment of the cache changes anything for us and change it. But IIRC this was mostly related to
binaries installed by cargo.

For the install rust action, we can exchange this with a script installing rustc & cargo, it doesn't look like there's an official setup-rust action like for other compilers. It is a bit messy and requires a curl pipe shell but eliminates the dependency here (though the actions-rs are defacto official as it is used by the rust team themselves):

```yml
- name: Install Rust
  run: |
    curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs/ | sh -s -- -y
    source $HOME/.cargo/env
    rustup toolchain install --profile minimal --force stable
```

I'll add a commit for both.